### PR TITLE
Feature/#182 refactor gamemanager

### PR DESCRIPTION
**Pull request by leroykorterink**

### CODE REVIEW

File: src/Assets/LightSkyDefense/Scripts/GameStates/GameState.cs [hunk 1]
```diff
@@ -3,10 +3,33 @@
...
+    [HideInInspector]
+    public GameManager GameManager;
 
+    /// <summary>
+    /// Get GameManager for later use
+    /// </summary>
     private void Awake()
     {
-        _gameManager = GetComponent<GameManager>();
+        GameManager = GetComponent<GameManager>();
```
Comment (hassanAlAwwadi) [inline src/Assets/LightSkyDefense/Scripts/GameStates/GameState.cs]:
```suggestion
        GameManager = GameManager.Instance;
```

**Review by hassanAlAwwadi — DISMISSED** [review]

**Review by hassanAlAwwadi — DISMISSED** [review]

File: src/Assets/LightSkyDefense/Scripts/Common/Steering/Behaviours/FollowPath.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    public float FollowSpeed = .1f;
+
+    // The target position on the path
+    private Vector3 _progress;
+
+    // The waypoint index to which _progress moves towards
+    private int _currentWaypointIndex;
+
+    /// <summary>
+    /// Rigidbody is used to progress along the path using the velocity's magnitude
```
Comment (Maxboef) [inline src/Assets/LightSkyDefense/Scripts/Common/Steering/Behaviours/FollowPath.cs:25]: Summary?

File: src/Assets/LightSkyDefense/Scripts/Enemies/Abilities/FindTarget.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    private void Awake()
+    {
+        // Disable shooting state on awake by default
+        ShootAtTarget.enabled = false;
+    }
+
+    private void FixedUpdate()
+    {
+        var TargetsInRange = Physics.OverlapSphere(transform.position, Radius, LayerBitMask);
```
Comment (Maxboef) [inline src/Assets/LightSkyDefense/Scripts/Enemies/Abilities/FindTarget.cs:23]: Divide to multiple lines?

File: src/Assets/LightSkyDefense/Scripts/Enemies/Abilities/EnemyAoEHeal.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
+﻿using UnityEngine;
+
+public class EnemyAoEHeal : MonoBehaviour
+{
+    public float HealAmount = 0.1f;
```
Comment (Maxboef) [inline src/Assets/LightSkyDefense/Scripts/Enemies/Abilities/EnemyAoEHeal.cs:6]: Tooltips?

**Review by Maxboef — APPROVED** [review]

**Pull request merged**